core/trino-main/src/main/java/io/trino/sql/planner/LocalExecutionPlanner.java

```diff
@@ -3142,8 +3150,12 @@ public PhysicalOperation visitRefreshMaterializedView(RefreshMaterializedViewNod
         @Override
         public PhysicalOperation visitTableWriter(TableWriterNode node, LocalExecutionPlanContext context)
         {
+            int driverInstanceCount = getTaskWriterCount(session);
```

Comment:
My previous comment must've been confusing. Essentially what I was trying to suggest is to separate these two properties.

Ideally it would be great not to introduce an additional property defining the number of task writers. The `task.writer-count` would define the total number of writers allowed. If the writer scaling is disabled the system would always use the number of writers defined. If the writer scaling is enabled the system would gradually add more writers as necessary up to the `task.writer-count`.

If there's a good reason it is better not to rely on the previous property (e.g.: we don't want to change the default behavior) then it might make sense to add a separate property. However since the new functionality is enabled by default anyway I wonder if it makes sense to simply increase the `task.writer-count` to `8` and reuse the existing property (maybe mentioning in the release notes this change including a short instruction of how to revert the previous behavior).

Reply:
> However since the new functionality is enabled by default anyway I wonder if it makes sense to simply increase the task.writer-count to 8 and reuse the existing property (maybe mentioning in the release notes this change including a short instruction of how to revert the previous behavior).

I had a discussion around this with @raunaqmorarka. It seems like we can't easily change the default to 8 because scaling won't work for connectors which don't report physical written bytes, so changing to 8 could lead to a lot of small files. That's why I've created a separate config `minScaledWriterCount` which basically means that at-least scaling will happen to this count even when `task_writer_count` is small.

Reply:
Oh, that's a good point.